ksqldb-common/src/main/java/io/confluent/ksql/logging/processing/ProcessingLoggerFactory.java

```diff
@@ -25,6 +27,16 @@ public interface ProcessingLoggerFactory {
    */
   ProcessingLogger getLogger(String name);
 
+  /**
+   * Get a processing logger for writing record processing log messages. This processing logger
+   * also emits metrics
+   * @param name The name of the logger to get.
+   * @param metrics The metrics that are emitted.
+   * @param customMetricsTags A map of metrics tags.
+   * @return The logger with the given name that emits metrics.
+   */
+  ProcessingLogger getLoggerWithMetrics(String name, Metrics metrics, Map<String, String> customMetricsTags);
```

Comment:
Can we pass the `Metrics` object into the logger factory constructor? This way you don't need to pass it into every call that creates a logger. And, you can also use metrics from logs returned by `getLogger` - they just wouldn't have all the tags.

Reply:
done